File: core/src/androidMain/kotlin/ru/mobileup/template/core/map/internal/subcontrollers/ThemeController.kt
package ru.mobileup.template.core.map.internal.subcontrollers

import com.yandex.mapkit.mapview.MapView
import ru.mobileup.template.core.map.MapTheme

internal class ThemeController(
    private val mapView: MapView
) {
    fun setTheme(theme: MapTheme) {
        mapView.mapWindow.map.setMapStyle(theme.yandexMapStyleJson)
    }
}

File: iosApp/iosApp/Source/Map/Subcontrollers/ThemeController.swift
import YandexMapsMobile
import Shared

final class ThemeController {

    private let mapView: YMKMapView

    init(mapView: YMKMapView) {
        self.mapView = mapView
    }

    func setTheme(theme: MapTheme) {
        mapView.mapWindow.map.setMapStyleWithStyle(theme.yandexMapStyleJson)
    }
}
